app/status.go

```diff
@@ -281,6 +304,38 @@ func (a *App) SetStatusAwayIfNeeded(userId string, manual bool) {
 	a.SaveAndBroadcastStatus(status)
 }
 
+func (a *App) SetStatusDoNotDisturbTimed(userId string, endtime string) {
+	if !*a.Config().ServiceSettings.EnableUserStatuses {
+		return
+	}
+
+	status, err := a.GetStatus(userId)
+
+	if err != nil {
+		status = &model.Status{UserId: userId, Status: model.STATUS_OFFLINE, Manual: false, LastActivityAt: 0, ActiveChannel: ""}
+	}
+
+	status.PrevStatus = status.Status
+	status.Status = model.STATUS_DND
+	status.Manual = true
+
+	tz, aErr := a.Srv().Store.User().GetTimezone(userId)
+	if aErr != nil {
+		mlog.Error("Failed to fetch user's timezone", mlog.String("user_id", userId), mlog.String("err", aErr.Error()))
+	}
+
+	utc, cErr := convertToUTC(tz, endtime)
+
+	if cErr != nil {
+		mlog.Error("Failed to conver user timezone to utc", mlog.String("user_id", userId), mlog.String("err", cErr.Error()))
+		return
+	}
```

Comment:
I thought we agreed to send the time in UTC from the client itself? There is no need to do a conversion at the server side.

Reply:
I thought it'd be better at server side as it'll be easier to test and verify and should be faster than js.

Reply:
At server side, it requires an extra DB call. And bear in mind that the server is a common thing for all users, whereas a user's browser is just one. I think it's cleaner to get the right data from the API itself. So the server code looks cleaner and simpler.